john-deere-api.py

```diff
@@ -97,7 +98,8 @@ def needs_organization_access():
     for org in api_response['values']:
         for link in org['links']:
             if link['rel'] == 'connections':
-                return link['uri']
+                connectionsUri = link['uri']
+                return '%s?%s' % (connectionsUri, urllib.parse.urlencode({'redirect_uri': settings['orgConnectionCompletedUrl']}))
```

Comment:
```suggestion
                query = urllib.parse.urlencode({'redirect_uri': settings['orgConnectionCompletedUrl']})
                return f"{connectionsUri}?{query}"
```

Reply:
I think general concensus is `expr % params << expr.format(params) << f"{params}"`